lib/api/data-contracts.ts

```diff
@@ -266,6 +272,12 @@ export type TimedRunViewModel = BaseRunViewModel & {
   time: string
 }
 
+export interface UpdateLeaderboardRequest {
+  name?: string
+  slug?: string
+  info?: string
+}
```

Comment:
So this request rejects if we've got all fields empty. Would we want to do this instead?

```ts
export type UpdateLeaderboardRequest = {
  name: string,
  slug?: string,
  info?: string
} | {
  name?: string,
  slug: string,
  info?: string
} | {
  name?: string,
  slug?: string,
  info: string
}
```

Probably too wordy though, right

Reply:
Ideally this is something we'd just have to handle from wherever we make this request. Currently the `lib/api` folder and its contents are completely auto-generated by an external package. So anything we'd change here would have to be kept manually up to date on each update